polkadot/runtime/parachains/src/assigner_bulk/mod.rs

```diff
@@ -418,18 +429,37 @@ impl<T: Config> Pallet<T> {
 		assignments: Vec<(CoreAssignment, PartsOf57600)>,
 		end_hint: Option<BlockNumberFor<T>>,
 	) -> Result<(), DispatchError> {
-		// There should be at least one assignment
+		// TODO: Add this assert once the calls `request_core_count` and `notify_core_count`
+		// have been established. assert!(core < legacy_core_count + core_count);
+
+		// Begin should be no less than the Relay-chain block number on arrival of
+		// the message plus 10
+		let now = <frame_system::Pallet<T>>::block_number();
+		ensure!(begin >= now + BlockNumberFor::<T>::from(REQUIRED_LEAD_TIME), Error::<T>::InsufficientLeadTime);
```

Comment:
I don't think we should fail on this. We should just handle it best-effort. The RFC only states that 10 blocks should suffice for expected behavior. In practice less will do as well. Even past blocks would work, it is just that assignments would no longer be fully served.

In other words this is something the coretime chain should check, so it lives up to its side of expectations. The most important reason for the 10 block limit is certainly that processing of messages might be delayed. Therefore this check might fail, even if the coretime chain did nothing wrong: Sends 10 blocks in advance, we have a queue of messages and only process that message two blocks later -> Check fails, although there is absolutely no reason why this would not be valid.